pulpanalytics/management/commands/summarize.py

```diff
@@ -47,18 +41,24 @@ def _get_next_date_to_summarize(cls):
     @staticmethod
     def _handle_online_workers(systems, summary):
         online_workers_qs = OnlineWorkers.objects.filter(system__in=systems)
-        online_workers_stats = online_workers_qs.aggregate(Avg('processes'), Avg('hosts'))
+        try:
+            online_workers_stats = online_workers_qs.aggregate(Avg('processes'), Avg('hosts'))
 
-        summary.online_workers.processes__avg = online_workers_stats['processes__avg']
-        summary.online_workers.hosts__avg = online_workers_stats['hosts__avg']
+            summary.online_workers.processes__avg = online_workers_stats['processes__avg']
+            summary.online_workers.hosts__avg = online_workers_stats['hosts__avg']
+        except TypeError:
```

Comment:
What situation is this handling?

Reply:
This is being graceful with me having bad database records. (Systems without any other information)

Reply:
I'd prefer we not add code that silences a data quality problem. If for some reason the production site was in this situation I'd like us to find it via a log error. Can this be changed?